simulator/lang/lib.mjs

```diff
@@ -0,0 +1,264 @@
+import * as c from '@reach-sh/simulator-client';
+
+// nodejs compatible import
+// const c = await import('@reach-sh/simulator-client');
+
+// TODO: typescript
+
+const consensusID = -1
+const nwToken = -1
+
+Object.filter = (obj, predicate) =>
+  Object.fromEntries(Object.entries(obj).filter(predicate));
+
+
+class Scenario {
+  constructor() {
+    this.stateID = 0
+    this.participants = [];
+    this.consensus = new Consensus(new Account(-1),this);
+    this.apis = [];
+    this.views = [];
+  }
+
+  async init() {
+    // reset server
+    await c.resetServer()
+    // load the Reach program
+    const rsh = await c.load()
+    // initialize the program for the Consensus
+    await c.init()
+    const apis = await c.getAPIs()
+    const views = await c.getViews(this.stateID)
+    const l = await c.getStateLocals(this.stateID)
+
+    // setup parts
+    for (const [k,v] of Object.entries(l.l_locals)) {
+      const who = v.l_who ? v.l_who : 'Consensus'
+      const acc = new Account(v.l_acct)
+      if (who) {
+        const p = new Participant(k,acc,who,this)
+        this.participants.push(p)
```

Comment:
They should at least be indexed by their name, not position

Reply:
yeah